packages/babel-core/src/config/files/module-types.ts

```diff
@@ -134,6 +135,22 @@ async function loadMjsDefault(filepath: string) {
 
 function getTSPreset(filepath: string) {
   try {
+    const packageJson = require("@babel/preset-typescript/package.json");
+
+    if (
+      semver.lte(
```

Comment:
I would describe previous versions 7.21.0 as "they supports cts files, but it has a bug that makes it not work with imports/exports". This would also avoid the breakage of throwing for whoever is _already_ using `babel.config.cts` files with an older version of the preset (which works for all the simple self-contained configs)